### PR TITLE
edward: round11 — model_dropout=0.05 (regularization, single delta)

### PR DESCRIPTION
**Pull request by morganmcg1**
## Hypothesis

The SOTA config uses `model_dropout=0.0` (default — no dropout). Transolver's slice attention operates over a 65k-point surface + 65k-point volume input with only 400 training cases. With this data regime the model may overfit to specific geometric features. Small dropout (0.05 = 5%) on the attention projections could act as stochastic regularization, smoothing the generalization gap between val (34 cases) and test (50 cases).

Evidence supporting this direction: the askeladd #141 compound run showed val −0.41% but test +0.23% (val/test ratio 1.123 vs SOTA's 1.115) — consistent with mild overfitting to val. Dropout is a standard fix for this class of problem.

**Single delta:** `--model-dropout 0.05` added to SOTA command, everything else unchanged.

**Risk:** Dropout adds noise to gradient updates; in Lion's sign-gradient regime this could destabilize early training. Monitor ep1–2 trajectory vs SOTA (target: stay within +10% of SOTA at ep2 = 24.15).

## Instructions

Single change from SOTA: add `--model-dropout 0.05` to the SOTA command.

**Exact command:**
```bash
cd target/
torchrun --standalone --nproc_per_node=8 train.py \
  --agent edward \
  --optimizer lion --lr 1e-4 --weight-decay 5e-4 \
  --lion-beta1 0.9 --lion-beta2 0.99 \
  --no-compile-model --batch-size 4 --validation-every 1 \
  --train-surface-points 65536 --eval-surface-points 65536 \
  --train-volume-points 65536 --eval-volume-points 65536 \
  --model-layers 4 --model-hidden-dim 512 --model-heads 8 --model-slices 128 \
  --model-dropout 0.05 \
  --use-ema --ema-decay 0.999 \
  --wandb-group tay-round11-model-dropout-0p05 \
  --wandb-name edward/round11-model-dropout-0p05
```

This is the **only** change vs SOTA `d03oghpp`: `--model-dropout 0.05` (SOTA default is 0.0, no dropout).

Note: Do **not** add `--lr-cosine-t-max` — SOTA `d03oghpp` used `lr_cosine_t_max=0` (flat LR, no cosine).

## Baseline

Current SOTA — PR #115 (thorfinn, lr=1e-4 + EMA=0.999), test_abupt **10.580**, val 9.484 (ep9), W&B run `d03oghpp`.

| Metric | PR #115 SOTA | AB-UPT ref |
|---|---:|---:|
| `abupt` | **10.580** | — |
| `surface_pressure` | 5.690 | 3.82 |
| `wall_shear` | 10.419 | 7.29 |
| `volume_pressure` | 12.740 | 6.08 |
| `tau_x` | 8.908 | 5.35 |
| `tau_y` | 12.491 | 3.65 |
| `tau_z` | 13.071 | 3.63 |

**Reference val trajectory (SOTA):** 53.75 / 24.15 / 16.51 / 13.47 / 11.83 / 10.88 / 10.16 / 9.73 / 9.48

## Results

_(fill in after training completes)_

Post: best `val_primary/abupt_axis_mean_rel_l2_pct`, final `test_primary/abupt_axis_mean_rel_l2_pct`, per-component test breakdown, W&B run URL.

### CODE REVIEW

**Comment by morganmcg1** [comment]
STUDENT edward:

## Results

**TL;DR:** `--model-dropout 0.05` is a clear regression vs SOTA. Test abupt **11.029** vs SOTA 10.580 (**+4.24%**). Every test axis got worse. Val was also worse from epoch 1 onward — dropout slowed convergence rather than reducing overfit. The val/test ratio did tighten very slightly (1.105 vs SOTA 1.115), so the overfit hypothesis isn't strictly wrong, but dropout 0.05 over-regularizes for this small-data regime. Hypothesis rejected.

### Test metrics (best-EMA checkpoint, ep9)

| Metric | this run (dropout=0.05) | SOTA d03oghpp | Δ vs SOTA |
|---|---:|---:|---:|
| `abupt` | **11.029** | 10.580 | **+0.449 (+4.24%)** |
| `surface_pressure` | 6.062 | 5.690 | +0.372 (+6.54%) |
| `wall_shear` | 10.930 | 10.419 | +0.511 (+4.90%) |
| `volume_pressure` | 12.902 | 12.740 | +0.162 (+1.27%) |
| `tau_x` | 9.322 | 8.908 | +0.414 (+4.65%) |
| `tau_y` | 13.178 | 12.491 | +0.687 (+5.50%) |
| `tau_z` | 13.682 | 13.071 | +0.611 (+4.68%) |

Best val abupt = **9.977** at ep9 (vs SOTA 9.484, **+5.20%** regression on val too).

### Val trajectory comparison

| Epoch | this run | SOTA d03oghpp | Δ |
|---:|---:|---:|---:|
| 1 | 57.16 | 53.75 | +6.34% |
| 2 | 25.51 | 24.15 | +5.62% |
| 3 | 16.95 | 16.51 | +2.65% |
| 4 | 13.76 | 13.47 | +2.18% |
| 5 | 12.23 | 11.83 | +3.40% |
| 6 | 11.32 | 10.88 | +4.04% |
| 7 | 10.67 | 10.16 | +5.05% |
| 8 | 10.14 | 9.73 | +4.21% |
| 9 | 9.98 | 9.48 | +5.20% |

Both runs hit the 270-min train budget at exactly 9 val epochs. Dropout did not extend training, just slowed per-epoch progress in equal wall time.

### Generalization-gap check (the hypothesis)

| | val (best) | test | val/test ratio |
|---|---:|---:|---:|
| dropout=0.05 (this) | 9.977 | 11.029 | **1.105** |
| SOTA d03oghpp | 9.484 | 10.580 | **1.115** |

Ratio tightened by only 0.010 (≈0.9% relative). The PR cited askeladd #141 ratio at 1.123. Dropout 0.05 buys a tiny generalization-gap improvement but at the cost of ~5% absolute regression on every metric. Net effect is negative.

### Run details

- **W&B run (rank 0):** `e5l1r38b` — group `tay-round11-model-dropout-0p05`
  https://wandb.ai/wandb-applied-ai-team/senpai-v1-drivaerml-ddp8/runs/e5l1r38b
- **Runtime:** 270.76 min (matched SOTA budget)
- **Best epoch:** 9 (EMA checkpoint)
- **Peak GPU mem:** ~75 GB / 96 GB per GPU (allocated ≈73%) — no OOM risk
- **Config:** Lion lr=1e-4 wd=5e-4 (β1=0.9, β2=0.99), batch=4, model 4L/512d/8h/128slices, EMA=0.999, **dropout=0.05** (single delta), volume_loss_weight=1.0 (matches SOTA), no compile, lr_cosine_t_max=0 (flat LR), warmup=0
- **Exact command:**
  ```bash
  cd target/
  torchrun --standalone --nproc_per_node=8 train.py \
    --agent edward \
    --optimizer lion --lr 1e-4 --weight-decay 5e-4 \
    --lion-beta1 0.9 --lion-beta2 0.99 \
    --no-compile-model --batch-size 4 --validation-every 1 \
    --train-surface-points 65536 --eval-surface-points 65536 \
    --train-volume-points 65536 --eval-volume-points 65536 \
    --model-layers 4 --model-hidden-dim 512 --model-heads 8 --model-slices 128 \
    --model-dropout 0.05 \
    --use-ema --ema-decay 0.999 \
    --wandb-group tay-round11-model-dropout-0p05 \
    --wandb-name edward/round11-model-dropout-0p05
  ```

### What happened

Dropout 0.05 on the slice-attention projections injected enough noise to slow convergence under Lion's sign-gradient updates without delivering a meaningful regularization payoff. Symptoms:

1. **Slower convergence, equal budget.** Val ep1 was +6.3% worse and the gap never fully closed in 9 val epochs / 270 min. Both runs got 9 val epochs; the dropout model just spent each epoch reaching a worse point.
2. **Train loss did stay marginally higher.** Train summary base_mse_loss across ranks ~0.022–0.027 (vs SOTA's tighter cluster) — consistent with stochastic noise in the gradient. So the regularizer is "doing something," but not in a useful direction here.
3. **Val/test gap barely budged.** The whole motivation for trying dropout was to close a ~1.12 val/test ratio. We moved it to 1.105. That's a 1% relative improvement on the gap, paid for with a 5% regression on the absolute metric. Bad trade.
4. **All seven test axes regressed.** No metric improved — including volume_pressure which was already the weakest axis and where I'd most expected dropout to help (deepest network of slice attention layers operating on coarse 65k volume points).

The hypothesis "Transolver overfits this 400-case regime and dropout will help" appears to be **wrong at this layer / location**. The askeladd #141 evidence (val −0.41% / test +0.23%) was a small, possibly noise-level signal, not a robust overfit. The model is closer to under-fit-given-budget than overfit, given that even SOTA's val curve was still descending at ep9.

### Suggested follow-ups

- **Smaller dropout (0.01–0.02).** If we still believe in mild overfit, 0.05 may be too strong. A 0.01 ablation would tell us whether *any* nonzero dropout helps. Low priority — this run suggests the marginal payoff is negative.
- **Dropout only after warmup.** Dropout from step 0 plus Lion's sign gradient could be hurting early-epoch optimization specifically. A `dropout_start_epoch=2` warmup variant would isolate "regularization vs early-training noise" effects. Medium effort (small change to model.py).
- **Stochastic depth / DropPath instead of attention dropout.** DropPath drops residual branches probabilistically and tends to be a better regularizer for transformer-style stacks than per-element attention dropout. Drop rate 0.05–0.1 on layer outputs only.
- **Weight decay sweep instead.** If overfit *is* a concern, scaling wd from 5e-4 to 1e-3 or 2e-3 (with `--weight-decay`) is a more orthogonal regularizer that doesn't interfere with Lion's gradient flow. Single-flag, easy to test.
- **Don't pursue this direction further at 65k points.** Given that even SOTA val is still falling at ep9 inside 270 min, the model is compute-limited, not overfit-limited. Better levers: longer training (extend timeout), data augmentation, or wider model.

### Honest call

Hypothesis rejected. SOTA stands at PR #115. Recommend closing this PR as a negative result; a 0.01–0.02 dropout sweep could be queued as a follow-up if the advisor wants a tighter answer, but I'd prioritize wd-sweep or DropPath ahead of it.

**Comment by morganmcg1** [comment]
ADVISOR: Closing PR #162 because ADVISOR: Negative result — model_dropout=0.05 regresses all metrics across the board (test_abupt +4.24%, val +5.20% vs SOTA). Edward's analysis is thorough and the conclusion is correct: the model is compute-limited not overfit-limited at 270-min budget. The val/test gap closed by only 0.010 absolute — not worth a 5% regression. Closing as dead end. Good analysis, Edward — the suggested follow-ups (especially DropPath and wd sweep) are well-reasoned and have been noted for future rounds..
